app/src/main/java/org/cru/godtools/fragment/ToolsFragment.java

```diff
@@ -117,19 +138,30 @@ public void onCreate(@Nullable final Bundle savedInstanceState) {
             mMode = args.getInt(EXTRA_MODE, mMode);
         }
 
+        mSettings = Settings.getInstance(requireContext());
+
         startLoaders();
     }
 
     @Override
     public View onCreateView(@NonNull final LayoutInflater inflater, @NonNull final ViewGroup container,
                              @Nullable final Bundle savedInstanceState) {
-        return inflater.inflate(R.layout.fragment_tools, container, false);
+        mToolsBinding = FragmentToolsBinding.inflate(inflater, container, false);
+        return mToolsBinding.getRoot();
     }
 
     @Override
     public void onViewCreated(@NonNull final View view, @Nullable final Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
         setupToolsList();
+        mToolsBinding.setCallback(mTutorialCallbacks);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        mToolsBinding.setIsTutorialViewable(
+                !mSettings.isFeatureDiscovered(Settings.FEATURE_OPT_IN_TUTORIAL));
```

Comment:
you should be updating the visibility of the tool when the preference changes, not in `onResume()`